Assets/Scripts/Features/UniversalGroundChecker.cs

```diff
@@ -0,0 +1,118 @@
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Baz_geluk9.Feature_vault.Feature
+{
+    /// <summary>
+    /// Universal ground checker supporting 2D and 3D environments with line and sphere casts.
+    /// Supports both offset and transform-based ground checking.
+    /// </summary>
+    public sealed class UniversalGroundChecker : MonoBehaviour
+    {
+        [Header("Usage")]
```

Comment:
Suggestion: maybe wrap all serializable properties using regions? It provides no benefits except reducing clutter in the file